common/components/common/datetime/DateRangeSelectors.jsx

```diff
@@ -0,0 +1,116 @@
+// @flow
+
+import React from 'react';
+import DatePicker from "react-datepicker";
+
+
+type Props = {|
+  dateTimeStart: ?string,
+  dateTimeEnd: ?string,
+  onChangeTimeStart: (string) => null,
+  onChangeTimeEnd: (string) => null,
+  formLabels: ?$ReadOnlyArray<string>,
+  formIds: ?$ReadOnlyArray<string>
+|};
+
+type State = {|
+  dateTimeStart: string,
+  dateTimeEnd: string,
+  formLabels: $ReadOnlyArray<string>,
+  formIds: $ReadOnlyArray<string>
+|}
+export class DateRangeSelectors extends React.PureComponent<Props, State> {
+  constructor(props: Props): void {
+    super();
+    
+    this.config = {
+      timeIntervals: 15,
+      timeFormat: "HH:mm",
+      dateFormat: "MMMM d, yyyy h:mm aa"
+    };
+
+    this.state = {
+      dateTimeStart: props.dateTimeStart || new Date(),
+      dateTimeEnd: props.dateTimeEnd,
+      formLabels: props.formLabels || ["Start", "End"],
+      formIds: props.formIds || ["startDate", "endDate"]
+    };
+    
+    this.onChangeDateTime = this.onChangeDateTime.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps: Props): void {
+    this.setState({
+      dateTimeStart: nextProps.dateTimeStart,
+      dateTimeEnd: nextProps.dateTimeEnd
+    });
+  }
+ 
+  onChangeDateTime(start: string, end: string): void {
```

Comment:
Like we discussed, I'm not a fan of silently changing user input here, even if we think we're doing them a favor. I'd much prefer validation and returning an error over silently switching their inputs around.